src/dashboards/teachers/details.jsonnet

```
@@ -398,3 +398,415 @@ dashboard.new(
+    description: |||
+      The distribution of video quality chosen by the users.
+      By default, the video quality is set to `480`.
+    |||,
+    options: {
+      legend: {
+        displayMode: 'hidden',
+      },
+    },
+    fieldConfig: {
+      overrides: [
+        {
+          matcher: {
+            id: 'byName',
+            options: 'Unique Count',
+          },
+          properties: [
+            {
+              id: 'displayName',
+              value: 'Users',
+            },
+          ],
+        },
+      ],
+    },
+    targets: [
+      {
+        datasource: common.datasources.lrs,
+        query: '%(video_query)s AND verb.id:"interacted"' % {
+          video_query: teachers_common.queries.video_id,
+        },
+        metrics: [common.metrics.cardinality(common.fields.actor_account_name)],
+        bucketAggs: [
+          {
+            id: '2',
+            type: 'terms',
+            settings: {
+              min_doc_count: '1',
+              size: '10',
+              order: 'desc',
+              orderBy: '_term',
+            },
+            field: common.fields.quality,
+          },
+        ],
+        timeField: '@timestamp',
+      },
+    ],
+    datasource: common.datasources.lrs,
+    transformations: [
+      {
+        id: 'convertFieldType',
+        options: {
+          conversions: [
+            {
+              targetField: common.fields.quality,
+              destinationType: 'string',
+            },
+          ],
+        },
+      },
+      {
+        id: 'filterByValue',
+        options: {
+          filters: [
+            {
+              fieldName: common.fields.quality,
+              config: {
+                id: 'equal',
+                options: {
+                  value: '480',
+                },
+              },
+            },
+          ],
+          type: 'exclude',
+          match: 'all',
+        },
+      },
+    ],
+  },
+  gridPos={ x: 16, y: 26, w: 4, h: 6 }
+).addPanel(
+  {
+    type: 'barchart',
+    title: 'Video volume user choice',
+    description: |||
+      The number of users who chosed to reduce the video volume or completely mute the video.
+      By default, the volume is set to `1`. It can only be reduced.
+    |||,
+    options: {
+      legend: {
+        displayMode: 'hidden',
+      },
+    },
+    fieldConfig: {
+      overrides: [
+        {
+          matcher: {
+            id: 'byName',
+            options: common.fields.volume,
+          },
+          properties: [
+            {
+              id: 'mappings',
+              value: [
+                {
+                  type: 'value',
+                  options: {
+                    '0': {
+                      text: 'Mute',
+                      index: 1,
+                    },
+                  },
+                },
+                {
+                  type: 'range',
+                  options: {
+                    from: 0,
+                    to: 1,
```

Are range limits excluded? Or should we be more precise?

```suggestion
                    from: 0.051,
                    to: 0.99,
```

---

I have considered the edge cases. As there is another mappings in the override, the `0` value ovveride takes over the [0,1] range mapping and the transformation of excluding results associated to `1` takes over the range mapping too. In the production the volume is sometimes very precise, up to 3 decimals. Hence I though it was better to have the [0,1] and use the priority rule established by Grafana on the value mappings and transformation.